### PR TITLE
Test_run bug

### LINKED ISSUE #651: CLI test_run failing
CLI test_run <case_name> fails with: 
```(2020.01.rc0-py27) [oyveid@be-lx128374 poly_normal]$ ert test_run poly.ert
Traceback (most recent call last):
  File "/prog/res/komodo/2020.01.rc0-py27/root/bin/ert", line 5, in <module>
    main()
  File "/prog/res/komodo/2020.01.rc0-py27/root/lib/python2.7/site-packages/ert_shared/main.py", line 261, in main
    args.func(args)
  File "/prog/res/komodo/2020.01.rc0-py27/root/lib/python2.7/site-packages/ert_shared/cli/main.py", line 36, in run_cli
    if args.current_case:
AttributeError: 'Namespace' object has no attribute 'current_case'
```
Because `test_run` doesn't have current case.

### PR DESCRIPTION
**Pull request by oyvindeide**
Resolves #651

### CODE REVIEW

File: tests/cli/test_integration.py [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
+    mocked_thread_start = Mock()
+    mocked_thread_join = Mock()
+    monkeypatch.setattr(threading.Thread, "start", mocked_thread_start)
+    monkeypatch.setattr(threading.Thread, "join", mocked_thread_join)
+    monkeypatch.setattr(ert_shared.cli.monitor.Monitor, "monitor", mocked_monitor)
+    yield mocked_monitor, mocked_thread_join, mocked_thread_start
+
+
+@pytest.mark.usefixtures("source_root")
+def test_target_case_equal_current_case(tmpdir, source_root):
```
Comment (dotfloat) [inline tests/cli/test_integration.py]: The `@pytest.mark.usefixtures` lines are superfluous; pytest will automatically figure out that you want to use the `source_root` fixture based on test arguments.

File: tests/all/test_main.py [hunk 2]
```diff
@@ -1,139 +1,141 @@
...
-        self.assertEquals(parsed.func.__name__, "run_cli")
-        self.assertFalse(parsed.verbose)
-
-
-if __name__ == '__main__':
-    unittest.main()
+if sys.version_info >= (3, 3):
+    from unittest.mock import Mock
+else:
+    from mock import Mock
```
Comment (dotfloat) [inline tests/all/test_main.py]: Do we need `unittest.mock`'s version of `Mock`? Having 4 lines for an import is fine if it happens once, but I can imagine we'd want to use `Mock` in other tests later on, where it'd be easier to use `from mock import Mock` and later change to `from unittest.mock import Mock` when we drop all pythons <3.3.

File: tests/all/test_main.py [hunk 3]
```diff
@@ -1,139 +1,141 @@
...
-    def test_argparse_exec_ensemble_experiment_current_case(self):
-        parser = ArgumentParser(prog="test_main")
-        parsed = ert_parser(parser, [ENSEMBLE_EXPERIMENT_MODE, "--current-case", 'test_case',
-                                     'test-data/local/poly_example/poly.ert'])
-        self.assertEquals(parsed.mode, ENSEMBLE_EXPERIMENT_MODE)
-        self.assertEquals(
-            parsed.config, "test-data/local/poly_example/poly.ert")
-        self.assertEquals(parsed.current_case, 'test_case')
-        self.assertEquals(parsed.func.__name__, "run_cli")
-        self.assertFalse(parsed.verbose)
```
Comment (dotfloat) [inline tests/all/test_main.py]: Can't find this in new tests

File: tests/all/test_main.py (hunk 2 above)
Comment (oyvindeide) [reply tests/all/test_main.py]: Will become an issue of semantics, but doing it this way means we only have to delete everything but `from unittest.mock import Mock` when the day comes, won't have to add anything.

File: tests/all/test_main.py (hunk 3 above)
Comment (oyvindeide) [reply tests/all/test_main.py]: Good catch, added now :+1:

**Pull request merged**